sdk/textanalytics/ai-text-analytics/src/models.ts

```diff
@@ -391,7 +443,512 @@ export interface Opinion {
+   */
+  fhirBundle?: { [propertyName: string]: any };
+}
+
+/**
+ * An error result from the healthcare operation on a single document.
+ */
+export type HealthcareErrorResult = TextAnalysisErrorResult;
+
+/**
+ * The result of the healthcare operation on a single document.
+ */
+export type HealthcareResult = HealthcareSuccessResult | HealthcareErrorResult;
+
+/**
+ * The result of the custom recognize entities operation on a single document.
+ */
+export type CustomEntityRecognitionResult =
+  | CustomEntityRecognitionSuccessResult
+  | CustomEntityRecognitionErrorResult;
+
+/**
+ * The result of the recognize custom entities operation on a single document,
+ * containing a collection of the entities identified in that document.
+ */
+export interface CustomEntityRecognitionSuccessResult extends TextAnalysisSuccessResult {
+  /**
+   * The collection of entities identified in the input document.
+   */
+  entities: Entity[];
+}
+
+/**
+ * An error result from the recognize custom entities operation on a single document.
+ */
+export type CustomEntityRecognitionErrorResult = TextAnalysisErrorResult;
+
+/**
+ * The result of the custom classify document single category operation on a single document.
+ */
+export type CustomSingleLabelClassificationResult =
+  | CustomSingleLabelClassificationSuccessResult
+  | CustomSingleLabelClassificationErrorResult;
+
+/**
+ * The result of the custom classify document single category operation on a single document,
+ * containing the result of the classification.
+ */
+export interface CustomSingleLabelClassificationSuccessResult extends TextAnalysisSuccessResult {
+  /**
+   * The collection of classifications in the input document.
+   */
+  classifications: ClassificationCategory[];
+}
+
+/**
+ * An error result from the custom classify document single category operation on a single document.
+ */
+export type CustomSingleLabelClassificationErrorResult = TextAnalysisErrorResult;
+
+/**
+ * The result of the custom classify document multi categories operation on a multi document.
+ */
+export type CustomMultiLabelClassificationResult =
+  | CustomMultiLabelClassificationSuccessResult
+  | CustomMultiLabelClassificationErrorResult;
+
+/**
+ * The result of the custom classify document multi categories operation on a multi document,
+ * containing the result of the classification.
+ */
+export interface CustomMultiLabelClassificationSuccessResult extends TextAnalysisSuccessResult {
+  /**
+   * The collection of classifications in the input document.
+   */
+  classifications: ClassificationCategory[];
+}
+
+/**
+ * An error result from the custom classify document multi category operation on a multi document.
+ */
+export type CustomMultiLabelClassificationErrorResult = TextAnalysisErrorResult;
+
+/**
+ * Options common to all batch actions.
+ */
+export interface AnalyzeBatchActionCommon {
+  /**
+   * The name of the action.
+   */
+  actionName?: string;
```

Comment:
can a user set this? or is it used internally?

Reply:
The customer can set it. I believe the service supports multiple tasks of the same kind now so it can be handy. Do you think there is any drawback to this?

Reply:
We use it internally in python to ensure the actions are ordered correctly with the results. Do you know if the service has guaranteed ordering for language API? If so, then I don't see a drawback to this.

Reply:
Abby confirmed a few weeks ago that the service guarantee action results ordering. @abhahn please correct me if I am wrong.

Reply:
I'm seeing what looks like out-of-order action results being returned in one of my recordings here: https://github.com/kristapratico/azure-sdk-for-python/blob/44c3eac5f19b47a2e9edcf23943221b513ef1256/sdk/textanalytics/azure-ai-textanalytics/tests/recordings/test_analyze.pyTestAnalyzetest_disable_service_logs.json

Abby to confirm, though.

Reply:
I see, it seems like task 9 is first in line but it should have been last. @abhahn is that a bug?

Reply:
Abby said she will look into it. In the meantime, I will keep `actionName` exposed. We can revisit this closer to the stable release if needed.